Stripe/StripeiOSTests/STPSourceFunctionalTest.m

```diff
@@ -623,18 +624,28 @@ - (void)testCreateSource_klarna {
     }];
     [self waitForExpectationsWithTimeout:TestConstants.STPTestingNetworkRequestTimeout handler:nil];
 }
-/*
-// Failing test, due to USD not being a valid currency:
- https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2454
+
+// 7/5/2023: Unfortunately we have to mock this test out because Wechat is strict about the usage of test merchants
+// in production.  We were previously using:
+// STPAPIClient *client = [[STPAPIClient alloc] initWithPublishableKey:@"pk_live_L4KL0pF017Jgv9hBaWzk4xoB"];
+// However, unable to continue with running this test live in production
 - (void)testCreateSource_wechatPay {
     STPSourceParams *params = [STPSourceParams wechatPayParamsWithAmount:1010
                                                                 currency:@"usd"
                                                                    appId:@"wxa0df51ec63e578ce"
                                                      statementDescriptor:nil];
 
-    STPAPIClient *client = [[STPAPIClient alloc] initWithPublishableKey:@"pk_live_L4KL0pF017Jgv9hBaWzk4xoB"];
+    id mockClient = OCMClassMock([STPAPIClient class]);
+    OCMStub([mockClient createSourceWithParams:[OCMArg any]
+                                    completion:[OCMArg any]])
+    .andDo(^(NSInvocation *invocation) {
+        void (^completion)(STPSource *data, NSError *error);
+        [invocation getArgument:&completion atIndex:3];
+        completion(STPFixtures.weChatPaySource, nil);
+    });
```

Comment:
I ask about testmode b/c if we mock this out I'm not sure if this ends up testing enough of our code.  Maybe we can run this in testmode and comment out the failing assert(s) to unblock for now.

Reply:
Hmmm -- How about we have both:  one test which has the mocked response which tests that we can properly deserialize `weChatAppURL`, and another non-mocked test using the publishable key?